lib/improver/nowcasting/forecasting.py

```diff
@@ -324,3 +327,101 @@ def grid_spacing(coord):
+                coordinate axis
+            vel_y (iris.cube.Cube):
+                Cube containing a 2D array of velocities along the y
+                coordinate axis
+
+
+        Keyword Args:
+            orographic_enhancement_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement fields. May have
+                data for multiple times in the cube. This is removed from the
+                input_cube before advecting, and added back on after advection.
+            metadata_dict (Dictionary):
+                Dictionary containing information for amending the metadata
+                of the output cube. Please see the
+                :func:`improver.utilities.cube_metadata.amend_metadata`
+                for information regarding the allowed contents of the metadata
+                dictionary.
+        """
+        self.orographic_enhancement_cube = orographic_enhancement_cube
+        if self.orographic_enhancement_cube:
+            input_cube, = ApplyOrographicEnhancement("subtract").process(
+                input_cube, self.orographic_enhancement_cube)
+        elif "precipitation_rate" in input_cube.name():
+            msg = ("For precipitation fields, orographic enhancement "
+                   "cube must be supplied.")
+            raise ValueError(msg)
+        self.input_cube = input_cube
+        self.advection_plugin = AdvectField(
+            vel_x, vel_y, metadata_dict=metadata_dict)
+
+    def __repr__(self):
+        """Represent the plugin instance as a string."""
+        result = ('<CreateExtrapolationForecast: input_cube = {}, '
+                  'orographic_enhancement_cube = {}, '
+                  'advection_plugin = {}>'.format(
+                      self.input_cube.name(),
+                      self.orographic_enhancement_cube.name(),
+                      self.advection_plugin))
+        return result
+
+    def extrapolate(self, leadtime_hours=None):
+        """
+        Produce a new forecast cube for the supplied lead time. Creates a new
+        advected forecast and then reapplies the orographic enhancement if it
+        is supplied.
+
+        Args:
+            leadtime_hours (float):
+                The forecast leadtime we want to generate a forecast for
+                in hours.
+
+        Returns:
+            advected_cube (iris.cube.Cube):
+                New cube with updated time and extrapolated data.  New data
+                are filled with np.nan and masked where source data were
+                out of bounds (ie where data could not be advected from outside
+                the cube domain).
+
+        Raises:
+            ValueError: If no leadtime_hours are provided.
+        """
+        if leadtime_hours is None:
+            message = ("leadtime_hours must be provided in order to produce an"
+                       " extrapolated forecast")
+            raise ValueError(message)
```

Comment:
Or KeyError?

Reply:
I'm not sure it is a key error:
```
exception KeyError
Raised when a mapping (dictionary) key is not found in the set of existing keys.
```

Reply:
My thinking was that the key was not included in the keyword-arguments dictionary passed in to the function.